lib/mobius/core/event.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Mobius.Core.Event do
+  @moduledoc false
+
+  @valid_names [
+    :READY,
+    :CHANNEL_CREATE,
+    :CHANNEL_UPDATE,
+    :CHANNEL_DELETE,
+    :CHANNEL_PINS_UPDATE,
+    :GUILD_CREATE,
+    :GUILD_UPDATE,
+    :GUILD_DELETE,
+    :GUILD_BAN_ADD,
+    :GUILD_BAN_REMOVE,
+    :GUILD_EMOJIS_UPDATE,
+    :GUILD_INTEGRATIONS_UPDATE,
+    :GUILD_MEMBER_ADD,
+    :GUILD_MEMBER_REMOVE,
+    :GUILD_MEMBER_UPDATE,
+    :GUILD_ROLE_CREATE,
+    :GUILD_ROLE_UPDATE,
+    :GUILD_ROLE_DELETE,
+    :INVITE_CREATE,
+    :INVITE_DELETE,
+    :MESSAGE_CREATE,
+    :MESSAGE_UPDATE,
+    :MESSAGE_DELETE,
+    :MESSAGE_DELETE_BULK,
+    :MESSAGE_REACTION_ADD,
+    :MESSAGE_REACTION_REMOVE,
+    :MESSAGE_REACTION_REMOVE_ALL,
+    :MESSAGE_REACTION_REMOVE_EMOJI,
+    :PRESENCE_UPDATE,
+    :TYPING_START,
+    :USER_UPDATE,
+    :VOICE_STATE_UPDATE,
+    :VOICE_SERVER_UPDATE,
+    :WEBHOOKS_UPDATE
+  ]
+  @valid_string_names Enum.map(@valid_names, &Atom.to_string/1)
+
+  @type names ::
```

Comment:
```suggestion
  @type name ::
```
(nit) Singular feels more appropriate for types.